File: server/src/main/java/com/repair/server/service/AuthService.java
package com.repair.server.service;

import com.repair.server.dto.*; // 偷懒写法：导入所有DTO

public interface AuthService {

    void register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    // === 必须补上这两个新方法的定义 ===

    void logout(String accessToken);

    RefreshTokenResponse refreshToken(RefreshTokenRequest request);
}

File: client/src/services/authService.ts
import { post } from '@/utils/api'
import { LoginRequest, LoginResponse, RefreshTokenRequest, RefreshTokenResponse } from '@/types'

// ==================== 登录接口 ====================
export const login = (params: LoginRequest): Promise<LoginResponse> => {
  return post<LoginResponse>('/auth/login', params)
}

// ==================== 刷新 Token 接口 ====================
export const refreshToken = (params: RefreshTokenRequest): Promise<RefreshTokenResponse> => {
  return post<RefreshTokenResponse>('/auth/refresh', params)
}

// ==================== 登出接口 ====================
export const logout = (): Promise<void> => {
  return post<void>('/auth/logout')
}
